lib/charms/postgresql_k8s/v0/postgresql.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2022 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""PostgreSQL helper class.
+
+The `postgresql` module provides methods for interacting with the PostgreSQL instance.
+
+Any charm using this library should import the `psycopg2` or `psycopg2-binary` dependency.
+"""
+import logging
+
+import psycopg2
+from psycopg2 import sql
+
+# The unique Charmhub library identifier, never change it
+LIBID = "24ee217a54e840a598ff21a079c3e678"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+
+logger = logging.getLogger(__name__)
+
+
+class PostgreSQLCreateDatabaseError(Exception):
+    """Exception raised when creating a database fails."""
+
+
+class PostgreSQLCreateUserError(Exception):
+    """Exception raised when creating a user fails."""
+
+
+class PostgreSQLDeleteUserError(Exception):
+    """Exception raised when deleting a user fails."""
+
+
+class PostgreSQLGetPostgreSQLVersionError(Exception):
+    """Exception raised when retrieving PostgreSQL version fails."""
+
+
+class PostgreSQL:
+    """Class to encapsulate all operations related to interacting with PostgreSQL instance."""
+
+    def __init__(
+        self,
+        host: str,
+        user: str,
+        password: str,
+        database: str,
+    ):
+        self.host = host
+        self.user = user
+        self.password = password
+        self.database = database
+
+    def _connect_to_database(self, database: str = None) -> psycopg2.extensions.connection:
+        """Creates a connection to the database.
+
+        Args:
+            database: database to connect to (defaults to the database
+                provided when the object for this class was created).
+
+        Returns:
+             psycopg2 connection object.
+        """
+        connection = psycopg2.connect(
+            f"dbname='{database if database else self.database}' user='{self.user}' host='{self.host}' password='{self.password}' connect_timeout=1"
+        )
+        connection.autocommit = True
+        return connection
```

Comment:
I'm a PG noob, so it's probably fine. 

Do you need to `close()` or `stop()` `cursor` or `connection` instances?